### PR TITLE
No response when submitting an answer on deleted question.

### PR DESCRIPTION
**Pull request by realARCHaim**
- https://trello.com/c/0AxwXxPS (new bug case discoreverd)

### CODE REVIEW

File: askup/static/assets/application.js [hunk 1]
```diff
@@ -205,6 +205,13 @@ function on_answer_success(data) {
         });
         $('.show-on-answered').slideDown();
         $('.hide-on-answered').slideUp();
+        return;
+    }
+
+    if ('redirect_url' in data) {
+        var callback = function() {window.location.href = data.redirect_url;};
+        var notification = data.notification || 'Error on question answer, redirecting...';
```
Comment (idegtiarov) [inline askup/static/assets/application.js]: This is a very strange notification I would change it to some meaningful one.

File: askup/templates/askup/question_answer_modal.html [hunk 2]
```diff
@@ -9,7 +9,7 @@ <h4 class="correct" id="your-answer-read-only"></h4>
                 <br>
             </div>
             <div class="hide-on-answered">
-                <form id="answer-form" method="post" action="{% url 'askup:question_answer' 1 %}">
+                <form id="answer-form" method="post" action="{% url 'askup:question_answer' 1 1 %}">
```
Comment (idegtiarov) [inline askup/templates/askup/question_answer_modal.html:12]: We've decided that using args name will be more clear to understand what params are sent.

File: askup/utils/views.py [hunk 3]
```diff
@@ -466,3 +466,83 @@ def pop_next_existent_question_id_from_list(question_ids):
...
+    Returns HttpResponseRedirect to the qset where question was located, to the next question
+    in the quiz or to the organizations list.
+    """
+    if is_quiz:
+        return get_json_redirect_next_quiz_question(request.user.id, qset_id, filter)
+
+    return get_json_redirect_qset(qset_id)
+
+
+def get_json_redirect_organizations_list():
```
Comment (idegtiarov) [inline askup/utils/views.py]: Where is this function used?

File: askup/utils/views.py [hunk 4]
```diff
@@ -466,3 +466,83 @@ def pop_next_existent_question_id_from_list(question_ids):
...
+    """
+    Return the json object with redirect to the organizations list.
+    """
+    notification = (
+        'This question was deleted since you\'ve opened it,' +
```
Comment (idegtiarov) [inline askup/utils/views.py]: Typo: add space after the coma.

File: askup/utils/views.py [hunk 5]
```diff
@@ -466,3 +466,83 @@ def pop_next_existent_question_id_from_list(question_ids):
...
+    )
+
+
+def get_json_redirect_next_quiz_question(user_id, qset_id, filter):
+    """
+    Return the json object with redirect to the next question in the quiz.
+    """
+    next_question_id = get_next_quiz_question(user_id, filter, qset_id, False)
+    notification = (
+        'This question was deleted since you\'ve opened it,' +
```
Comment (idegtiarov) [inline askup/utils/views.py]: ditto

File: askup/utils/views.py [hunk 6]
```diff
@@ -466,3 +466,83 @@ def pop_next_existent_question_id_from_list(question_ids):
...
+            'notification': notification,
+        }
+    )
+
+
+def get_json_redirect_next_quiz_question(user_id, qset_id, filter):
+    """
+    Return the json object with redirect to the next question in the quiz.
+    """
+    next_question_id = get_next_quiz_question(user_id, filter, qset_id, False)
```
Comment (idegtiarov) [inline askup/utils/views.py:489]: What about the case when next_question_id is None?

File: askup/utils/views.py [hunk 7]
```diff
@@ -466,3 +466,83 @@ def pop_next_existent_question_id_from_list(question_ids):
...
+        }
+    )
+
+
+def get_json_redirect_qset(qset_id):
+    """
+    Return the json object with redirect to the qset.
+    """
+    notification = (
+        'This question was deleted since you\'ve opened it,' +
```
Comment (idegtiarov) [inline askup/utils/views.py]: ditto

File: askup/utils/views.py [hunk 8]
```diff
@@ -466,3 +466,83 @@ def pop_next_existent_question_id_from_list(question_ids):
...
+            'notification': notification,
+        }
+    )
+
+
+def get_json_redirect_qset(qset_id):
+    """
+    Return the json object with redirect to the qset.
+    """
+    notification = (
```
Comment (idegtiarov) [inline askup/utils/views.py]: This notification text is almost the same in all function why don't move it to the module's variable?

File: askup/utils/views.py [hunk 9]
```diff
@@ -466,3 +466,83 @@ def pop_next_existent_question_id_from_list(question_ids):
...
+            'redirect_url': reverse('askup:qset', kwargs={'pk': qset_id}),
+            'notification': notification,
+        }
+    )
+
+
+def do_make_answer_form(request, question):
+    """Compose and return answer form."""
+    if request.method == 'POST':
+        return AnswerModelForm(request.POST or None, parent_qset_id=question.qset_id)
```
Comment (idegtiarov) [inline askup/utils/views.py]: Why do you exchange empty request.POST object with the None?

**Review by idegtiarov — CHANGES_REQUESTED** [review]

File: askup/views.py [hunk 10]
```diff
@@ -734,16 +727,22 @@ def start_quiz_all(request, qset_id):
...
 
-def get_quiz_question_redirect(next_question_id, filter):
+def get_quiz_question_redirect(qset_id, next_question_id, filter):
     """Get quiz question redirect."""
     return redirect(
         '{0}?filter={1}'.format(
-            reverse('askup:question_answer', kwargs={'question_id': next_question_id}),
+            reverse(
+                'askup:question_answer',
+                kwargs={
```
Comment (idegtiarov) [inline askup/views.py]: I would use one line for kwargs

**Review by idegtiarov — CHANGES_REQUESTED** [review]

File: askup/static/assets/application.js (hunk 1 above)
Comment (realARCHaim) [reply askup/static/assets/application.js]: done

File: askup/templates/askup/question_answer_modal.html (hunk 2 above)
Comment (realARCHaim) [reply askup/templates/askup/question_answer_modal.html:12]: this part of code is already deleted in the next PR. made this changes for the sake of compatibility.

File: askup/utils/views.py (hunk 3 above)
Comment (realARCHaim) [reply askup/utils/views.py]: removed as the redundant on

File: askup/utils/views.py (hunk 4 above)
Comment (realARCHaim) [reply askup/utils/views.py]: refactored

File: askup/utils/views.py (hunk 5 above)
Comment (realARCHaim) [reply askup/utils/views.py]: refactored

File: askup/utils/views.py (hunk 6 above)
Comment (realARCHaim) [reply askup/utils/views.py:489]: done

File: askup/utils/views.py (hunk 7 above)
Comment (realARCHaim) [reply askup/utils/views.py]: refactored

File: askup/utils/views.py (hunk 8 above)
Comment (realARCHaim) [reply askup/utils/views.py]: done

File: askup/utils/views.py (hunk 9 above)
Comment (realARCHaim) [reply askup/utils/views.py]: Seams like it is an excessive expression. Removed None.

File: askup/views.py (hunk 10 above)
Comment (realARCHaim) [reply askup/views.py]: done

File: askup/static/assets/application.js [hunk 11]
```diff
@@ -205,6 +205,13 @@ function on_answer_success(data) {
         });
         $('.show-on-answered').slideDown();
         $('.hide-on-answered').slideUp();
+        return;
+    }
+
+    if ('redirect_url' in data) {
+        var callback = function() {window.location.href = data.redirect_url;};
```
Comment (idegtiarov) [inline askup/static/assets/application.js]: One semicolon is redundant.

File: askup/templates/askup/question_answer_modal.html (hunk 2 above)
Comment (idegtiarov) [reply askup/templates/askup/question_answer_modal.html:12]: Ok, got it.

File: askup/utils/views.py (hunk 8 above)
Comment (idegtiarov) [reply askup/utils/views.py]: `notification = QUESTION_DELETED_TEXT.format('the correspondent subject...')` is fair enough multistring split is not needed here

File: askup/utils/views.py [hunk 12]
```diff
@@ -466,3 +467,66 @@ def pop_next_existent_question_id_from_list(question_ids):
...
+    """
+    next_question_id = get_next_quiz_question(user_id, filter, qset_id, False)
+
+    if next_question_id is None:
+        return get_json_redirect_qset(qset_id)
+
+    notification = QUESTION_DELETED_TEXT.format('the next one in the Quiz...')
+    redirect_url = reverse(
+        'askup:question_answer',
+        kwargs={
```
Comment (idegtiarov) [inline askup/utils/views.py]: One line kwargs is prefered

**Review by idegtiarov — CHANGES_REQUESTED** [review]

File: askup/utils/views.py (hunk 8 above)
Comment (realARCHaim) [reply askup/utils/views.py]: missed those brackets

File: askup/static/assets/application.js (hunk 11 above)
Comment (realARCHaim) [reply askup/static/assets/application.js]: done

File: askup/utils/views.py (hunk 12 above)
Comment (realARCHaim) [reply askup/utils/views.py]: done

**Review by idegtiarov — APPROVED** [review]

**Pull request merged**